torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -1432,10 +1444,61 @@ def codegen_multi_output(self, name, value):
             super().codegen_multi_output(name, value)
 
     def codegen_subgraph(self, subgraph, outer_inputs, outer_outputs):
-        raise NotImplementedError("Control flow NYI in C++ wrapper codegen.")
+        self.writeline(f"{self.comment} subgraph: {subgraph.name}")
+        for inner_input, outer_input in zip(subgraph.graph.graph_inputs, outer_inputs):
+            if config.abi_compatible:
+                self.writeline(f"AtenTensorHandle {inner_input}_handle;")
+                self.writeline(
+                    f"AOTI_TORCH_ERROR_CODE_CHECK(aoti_torch_copy_tensor({outer_input}, &{inner_input}_handle));"
```

Comment:
This is done (and the new ABI shim function `aoti_torch_copy_tensor` is added) to separate the ownership of the underlying `at::Tensor` instances between the inner and the outer scope: e.g., calling `.reset()` at the end of the codegened subgraph (inner scope) should decrement the refcount of the underlying storage, but not delete the storage, as it should still be accessible in the parent subgraph (outer scope).